### PR TITLE
JSX Components & demos

### PR DESCRIPTION
**Pull request by GlynnPhillips**
**Description**

Enable components to be built in JSX instead of handlebars and for the demos to still work

**Justification**

We (Profile team) need to add some new/variants of components for an experiment we plan to run to onboard users to instant alerts. Whilst this repo is entirely handlebars the wider estate within Customer Products is moving towards JSX, we therefore felt that it would make more sense for this work to be done in JSX.

I have made the changes to enable JSX components to be demos with an example component first as a talking point incase @Financial-Times/content-discovery are aware of something I have overlooked or have plans that this conflicts with.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -47,6 +47,7 @@
     "css-loader": "^0.23.1",
     "denodeify": "^1.2.1",
     "eslint": "6.5.1",
+    "eslint-plugin-react": "^7.33.1",
```
Comment (asugar13) [inline package.json:50]: **Praise**: Good call on adding a linter

File: demos/app.js [hunk 2]
```diff
@@ -1,3 +1,5 @@
+require('sucrase/register');
```
Comment (asugar13) [inline demos/app.js:1]: **Question:** This is required for JSX to work on this repo right?

**Review by asugar13 — COMMENTED** [review]
Nothing blocking me from approving this PR but I would let content disco approve it instead. Maybe they have considerations against jsx.

File: demos/templates/demo.html [hunk 3]
```diff
@@ -21,66 +21,86 @@
...
 		<h1>{{title}}</h1>
 
+
+		<section
+			id="example-component"
+			class="demo-section">
+			<div class="o-grid-row">
+
+				<div data-o-grid-colspan="12">
+					<h2>JSX components</h2>
```
[the remark this answers is not in the document]
Comment (GlynnPhillips) [reply demos/templates/demo.html:31]: I am not sure what you mean by this comment 😂 could you explain a little more please

File: demos/app.js (hunk 2 above)
Comment (GlynnPhillips) [reply demos/app.js:1]: Yes that right

**Comment by metart43** [comment]
Hey @GlynnPhillips, I have no objections towards the JSX components. They are more familiar to me than the handlebar's templating language. 

My only question - was there a specific reason why you have chosen React 16 over 17/18?

**Comment by GlynnPhillips** [comment]
@metart43 thats a great question that made me realise its not needed however I will start by answering the questions of why React 16.

I just ran `npm install react` which choose v16 - which is happening I believe to avoid conflicts with other dependencies such as pagekits `"@financial-times/dotcom-server-handlebars` which ships with React v16.

Realising this made me rethink whether I needed React as a dependency in this project and the answer is No. Anything that consumes the JSX components from this package will need to supply React in the same way that the demos supply react. So I ended up removing the react dependency

File: components/jsx/example/example.jsx [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import React from 'react';
```
Comment (gyss) [inline components/jsx/example/example.jsx:1]: should `react` be a peerDependency?

**Comment by GlynnPhillips** [comment]
> Just that I'm not sure if there would be any unexpected side effect (in this case adding sucrase)

Any side effects would only impact the demos (this will come back to bite me..) so I think low chance/impact even if it does

File: components/jsx/example/example.jsx (hunk 4 above)
Comment (GlynnPhillips) [reply components/jsx/example/example.jsx:1]: Yes thats a great point, that would be the best way highlight this. I will change that

Comment (GlynnPhillips) [reply components/jsx/example/example.jsx:1]: Done

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (GlynnPhillips) [reply package.json:50]: I think (I am not an expert on this) this is handled by eslint-config-next 
https://github.com/Financial-Times/eslint-config-next/blob/main/index.js#L72-L88

**Review by asugar13 — APPROVED** [review]
Approving this PR but can you please link it to a ticket in JIRA?

**Comment by GlynnPhillips** [comment]
Closing because https://github.com/Financial-Times/n-myft-ui/pull/590 replaces it

**Pull request closed without merging**